packages/lesswrong/components/Layout.tsx

```diff
@@ -225,10 +262,19 @@ class Layout extends PureComponent<LayoutProps,LayoutState> {
     // then it should.
     // FIXME: This is using route names, but it would be better if this was
     // a property on routes themselves.
+
+    const routeName = location.currentRoute?.name
     const standaloneNavigation = !location.currentRoute ||
       standaloneNavMenuRouteNames[forumTypeSetting.get()]
-        .includes(location.currentRoute.name)
+        .includes(routeName)
+    
+    const whiteBackground = ["posts.single", "events.single", "tagIndex", "donate", "about", "faq", "CommunityHome"].includes(routeName)
```

Comment:
Don't do this; instead, add a property to the route objects

Reply:
Using route-name strings creates differences between routes that aren't captured in the routes file, which is bad, because sometimes we want to make a route that's a variant of an existing route.

Reply:
Seems fine to do, but curious what distinction you're drawing between this and the standaloneNavMenuRouteNames, which I was modeling this after. (I think JP originally made those, and if it was a pattern we shouldn't follow seems fine)

Reply:
Rephrasing as a more concrete question: does this imply that you think we should also change how we implement standaloneNavMenuRouteNames?

Reply:
Pushed a fix for this.